TaskSharper.Calender.WPF/ViewModels/CalendarWeekViewModel.cs

```diff
@@ -1,45 +1,78 @@
 ﻿using System;
 using System.Collections.Generic;
 using System.Collections.ObjectModel;
-using System.Globalization;
 using System.Linq;
-using System.Text;
 using System.Threading.Tasks;
 using Prism.Events;
 using Prism.Mvvm;
 using Prism.Regions;
 using TaskSharper.Calender.WPF.Events;
 using TaskSharper.Calender.WPF.Events.Resources;
+using Prism.Commands;
 
 namespace TaskSharper.Calender.WPF.ViewModels
 {
-    public class CalendarWeekViewModel : BindableBase
+    public class CalendarWeekViewModel : BindableBase 
     {
         private readonly CalendarEventsService _service;
         private const int DAYS_IN_WEEK = 7;
         private IEventAggregator _eventAggregator;
 
+        public DelegateCommand NextCommand { get; set; }
+
+        public DelegateCommand PrevCommand { get; set; }
+
         public ObservableCollection<CalendarDateViewModel> DateHeaders { get; set; }
         public ObservableCollection<CalendarEventsViewModel> EventContainers { get; set; }
 
+        public int WeeklyOffset { get; set; } = 0;
+
+
         public CalendarWeekViewModel(CalendarEventsService service, IEventAggregator eventAggregator)
         {
+
+
+            NextCommand = new DelegateCommand(NextWeek);
+            PrevCommand = new DelegateCommand(PrevWeek);
             _service = service;
             _eventAggregator = eventAggregator;
             DateHeaders = new ObservableCollection<CalendarDateViewModel>();
             EventContainers = new ObservableCollection<CalendarEventsViewModel>();
             _eventAggregator.GetEvent<SpinnerEvent>().Publish(EventResources.SpinnerEnum.Show);
             InitializeViews();
+            GetCalelndarEvents();
+        }
+
+        private void NextWeek()
+        {
+            WeeklyOffset++;
+            ReconstructView();
+        }
 
+        private void PrevWeek()
+        {
+            WeeklyOffset--;
+            ReconstructView();
+        }
+        private void ReconstructView()
+        {
```

Comment:
Should we start the spinner while reconstructing?

Reply:
fixed